source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,187 @@
+# NixOS Configuration debugging using virtual machines
+
+Draft 2022.08.11
+
+## What will you learn?
+
+One of the most important features of NixOS is the ability to configure the entire system declaratively.
+This is done using a configuration file that specifies the entire system state, including the packages to be installed and the various system settings and options.[^wikinixos]
+This guide introduces the functionality of Nix Package Manager to test and debug NixOS configurations independent of a working NixOS installation.
+It serves as an introduction to Nix Package Manager's ability to create virtualized instances of NixOS.
+
+## What do you need?
+
+<!-- todo links -->
+
+- Basic knowledge of the Nix language.
+- A working installation of Nix Package Manager or NixOS.
+- nixpkgs path set in NIX_PATH 
+
+## What is a NixOS configuration
+
+A NixOS configuration is a declarative definition of a Linux installation that specifies which packages to install and how to configure them.
+It is written in the Nix language and uses the Nixpkgs repository, its library, packages and modules.
+<!-- [in my dream such a blob would appear as mediawiki style page-preview at mentions of Nix language]: The Nix expression language is a pure, lazy, functional language. Its main purpose is to describe packages, compositions of packages, and variability within packages.[^nixlang] -->
+<!-- todo links: ancors to introductions to library packages and modules are currently not in the manuals -->
+A NixOS configuration is a file that contains a function that is structured acording to the module<!-- todo link -->  convention.
+A NixOS configuration is a function that follows the convention of a module<!-- todo link to reference-->. <!-- remove one version -->
+
+<!-- [old version 2022.08.10 should be deleted] -->
+<!-- The Git repository Nixpkgs provides the necessary infrastructure and information needed to specify a NixOS configuration[^nixpkgs]. -->
+<!-- Nixpkgs contains a library, packages, and modules. -->
+<!-- The library provides the functional infrastructure to conveniently create packages. -->
+<!-- The packages defined in Nixpkgs provide build scripts needed to build software. -->
+<!-- The NixOS modules are a way to configure the system. [previous formulation] provide a configuration infrastructure to generate the system's configuration files. -->
+
+## Starting from default NixOS Configuration
+
+On NixOS, the configuration file is normally located at `/etc/nixos/configuration.nix`.
+A different location can be specified using the environment variable `$NIX_PATH`.
```

Comment:
Since we won't use it, it seems irrelevant.
```suggestion
```

Diataxis: [Ignore options and alternatives.](https://diataxis.fr/tutorials/#ignore-options-and-alternatives)